compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java

```diff
@@ -418,7 +429,8 @@ private BLangConstantValue calculateBooleanComplement(BLangConstantValue value)
     }
 
     private BLangConstantValue visitExpr(BLangExpression node) {
-        if (!node.typeChecked) {
+
+        if (!node.typeChecked && !this.semtypeEnabled) {
```

Comment:
Please create an issue to check on this before merging this branch to `master`. We need to make sure we don't break the compiler when `semtypeEnabled` is true and we perform the normal compilations.

Reply:
created in #32778